go/cli/mcap/cmd/cat.go

```diff
@@ -109,57 +110,80 @@ func printMessages(
 			}
 			continue
 		}
-		switch schema.Encoding {
-		case "ros1msg":
-			transcoder, ok := transcoders[channel.SchemaID]
-			if !ok {
-				packageName := strings.Split(schema.Name, "/")[0]
-				transcoder, err = ros.NewJSONTranscoder(packageName, schema.Data)
+		if schema == nil || schema.Encoding == "" {
```

Comment:
Thanks for fixing the bug – please make a separate PR for this though so it can be landed independently of the discussion around CBOR.

Reply:
merged https://github.com/foxglove/mcap/pull/857